handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -2661,4 +2672,36 @@ public String toString() {
                     '}';
         }
     }
+
+    private interface NativeSslException {
+        int errorCode();
+    }
```

Comment:
How about using an `abstract class` instead?

```java
private static abstract class NativeSslException extends SSLException {
    final int errorCode; // or maybe add a package-private getter
    NativeSslException(String reason, int errorCode) {
        super(reason);
        this.errorCode = errorCode;
    }
}

private static final OpenSslException extends NativeSslException {...}
```

Reply:
This will not work as I also need this for `SSLHandshakeException`

Reply:
Doh, never mind, then.